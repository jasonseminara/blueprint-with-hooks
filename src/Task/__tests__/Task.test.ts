import { TaskAdapter } from '..';

const backEndTask = {
  id: '1234',
  type: 'TASK',
  taskDetail: {
    foo: 34,
    bar: 56,
    baz: [
      'a','b', { c: 67 },
    ]
  }
};

const backEndReport = {
  id: '1XYZ - ALERT - USER STUFF',
  type: 'REPORT',
  reportDetail: {
    foo: 34,
    bar: 56,
    baz: [
      'a','b', { c: 99 },
    ]
  }
};

describe('Task', () => {
  it('Should convert a Task to FETask', () => {
    const hash = TaskAdapter.toFrontEnd(backEndTask as any);
    expect(hash).toMatchSnapshot();
  });

  it('Should convert a Report to FEReport', () => {
    const hash = TaskAdapter.toFrontEnd(backEndReport as any);
    expect(hash).toMatchSnapshot();
  });

  it('Should convert a FEReport to a BEReport', () => {
    const {reportDetail:detail, ...report} = backEndReport;
    const hash = TaskAdapter.toBackEnd({...report, detail, hash:'abc123'} as any);
    expect(hash).toMatchSnapshot();
  });
})