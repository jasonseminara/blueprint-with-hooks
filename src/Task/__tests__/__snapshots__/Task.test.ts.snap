// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Should convert a FEReport to a BEReport 1`] = `
Object {
  "id": "1XYZ - ALERT - USER STUFF",
  "reportDetail": Object {
    "bar": 56,
    "baz": Array [
      "a",
      "b",
      Object {
        "c": 99,
      },
    ],
    "foo": 34,
  },
  "type": "REPORT",
}
`;

exports[`Task Should convert a Report to FEReport 1`] = `
Object {
  "detail": Object {
    "bar": 56,
    "baz": Array [
      "a",
      "b",
      Object {
        "c": 99,
      },
    ],
    "foo": 34,
  },
  "hash": "e7800906f5fd665a45304329ff07e2714cdd263d",
  "id": "1XYZ - ALERT - USER STUFF",
  "type": "REPORT",
}
`;

exports[`Task Should convert a Task to FETask 1`] = `
Object {
  "detail": Object {
    "bar": 56,
    "baz": Array [
      "a",
      "b",
      Object {
        "c": 67,
      },
    ],
    "foo": 34,
  },
  "hash": "79d677d533c91dac9a1b9a5e0dceb8d8bb9b1748",
  "id": "1234",
  "type": "TASK",
}
`;
